artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerSessionImpl.java

```diff
@@ -1178,7 +1183,7 @@ public void deleteQueue(final SimpleString queueToDelete) throws Exception {
          throw new ActiveMQNonExistentQueueException();
       }
 
-      server.destroyQueue(unPrefixedQueueName, this, true, false, true);
+      server.destroyQueue(unPrefixedQueueName, enforceSecurity ? this : null, true, false, true);
```

Comment:
What does `null` do here? It is not very clear if you enforcing something that previously was enforce always. @jbertram

Reply:
If `enforceSecurity` is `true` then `this` is passed and security will be enforced. Otherwise `null` is passed and security is not enforced. Previously security was always enforced which would prevent the broker from cleaning up the subscription queue in certain situations.